### DISCUSSION TITLE
install from source

### DISCUSSION THREAD

**Discussion by Guang91** [mboquien/cigale#2] [Show and tell]
To install from source, you will need to 

1. Install dependencies, python>=3.8, astropy, numpy, scipy, sqlite, sqlalchemy, matplotlib, configobj, rich. 
2. Download and install Git LFS (https://git-lfs.github.com/)
3. Download CIGALE repository by "git clone https://gitlab.lam.fr/cigale/cigale.git" 
4. Under the downloaded CIGALE repository, switch to the develop branch by "git checkout develop", if you want to install the develop version
5. Make sure the repository is updated by "git pull"
6. Clean up the database from the previous installation by "rm -r pcigale/data/\*/" (note that this is only for the develop version; for other versions < 2022, use "rm pcigale/data/*.db")
7. Build the database by "python setup.py build"
8. Install CIGALE by "python setup.py develop"

C’est tout ! 
PS: You only need to do Steps 1-4 once; future updates only have to do 5-8.   
PPS: To install on a server (where you don't have written permission to the program directory), use "python setup.py develop --user" for a developer installation with the executables located in $HOME/.local/bin. Once added to $PATH, then CIGALE runs from a local installation.

Please ask below if you have any questions/issues.

**Comment by gtorrini** [comment]
Hi! I keep getting an error from numpy (see attached) when I run step 7. How should I proceed?
[cigale_setup_error.txt](https://github.com/mboquien/cigale/files/8848845/cigale_setup_error.txt)

**Reply by mboquien** [reply]
Hello. I assume you are using git? You need to install and setup git-lfs (https://git-lfs.github.com/). Once this done, you can simply switch to another branch (for instance develop) and switch back. Normally it should work after this.

**Reply by gtorrini** [reply]
It's working now. Thanks!

**Reply by maciejpkoprowski** [reply]
Hello. I had the same error (well, almost):

ValueError: cannot reshape array of size 3 into shape (2,3,31,3)

But switching between branches (develop to master and back) did not help. 

Any ideas please?

**Reply by maciejpkoprowski** [reply]
Error attached
[error.txt](https://github.com/mboquien/cigale/files/9401164/error.txt)

**Reply by mboquien** [reply]
Can you try `git lfs pull`?

**Reply by maciejpkoprowski** [reply]
Yes, that solved the problem. Thank you for a prompt reply.

**Comment by ItsnaKF** [comment]
Hello, I have done following this step. but I kept getting an error said `'Parameter' object has no attribute 'clone'` after editing the pcigale.ini and the input file as below. What should I do with this? thank you in advance
[cig88.txt](https://github.com/mboquien/cigale/files/9137815/cig88.txt)
[pcigale_ini.txt](https://github.com/mboquien/cigale/files/9137816/pcigale_ini.txt)

**Reply by mboquien** [reply]
Can you show the full backtrace please? Also what version of CIGALE and Python are you using? Thanks!

**Reply by ItsnaKF** [reply]
Thank you for quick response. I am sorry but it is my fault for not putting the radio modules even though I put the radio bands in the input file. It is worked already. I am not sure why the error just disappear after adding the radio module

**Comment by indebetouw** [comment]
Should I be concerned with the warnings:?

/lustre/naasc/sciops/comm/rindebet/miniconda3/envs/hyperiong/lib/python3.8/site-packages/setuptools/command/develop.py:40: EasyInstallDeprecationWarning: easy_install command is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` and ``easy_install``.
        Instead, use pypa/build, pypa/installer, pypa/build or
        other standards-based tools.

        See https://github.com/pypa/setuptools/issues/917 for details.
        ********************************************************************************

!!
  easy_install.initialize_options(self)

**Reply by mboquien** [reply]
It should not be an issue as long a you do not run python ≥ 3.12 I believe.
